src/oauth2-client/controller/collection.ts

```diff
@@ -73,8 +74,17 @@ class ClientCollectionController extends Controller {
       requirePkce: ctx.request.body.requirePkce ?? false,
     };
 
-    const client = await create(newClient, redirectUris);
-    ctx.response.body = hal.newClientSuccess(client, redirectUris, clientSecret);
+    // If client id already exists in DB, throw error.
+    try {
+      const client = await create(newClient, redirectUris);
+      ctx.response.body = hal.newClientSuccess(client, redirectUris, clientSecret);
+    } catch (error: any) {
+      if (error instanceof UniqueViolationError) {
```

Comment:
Better for the service to check for this error and throw `Conflict`. This is still a bit too much data access and DB concern being handled by the controller.

Reply:
Updated PR :)